packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanel.tsx

```diff
@@ -1040,79 +1068,92 @@ export function TransferPanel() {
-      if (isSmartContractWallet && !selectedToken) {
-        return TransferPanelMainErrorMessage.SC_WALLET_ETH_NOT_SUPPORTED
-      }
+    const ethBalanceFloat = isDepositMode
+      ? ethL1BalanceFloat
+      : ethL2BalanceFloat
+
+    const selectedTokenBalanceFloat = isDepositMode
+      ? selectedTokenL1BalanceFloat
+      : selectedTokenL2BalanceFloat
+
+    // No error while loading ETH balance
+    if (ethBalanceFloat === null) {
+      return undefined
+    }
 
-      if (
-        isDepositMode &&
-        selectedToken &&
-        isWithdrawOnlyToken(selectedToken.address, l2Network.id)
-      ) {
+    // ERC-20
+    if (selectedToken) {
+      if (isDepositMode && selectedTokenIsWithdrawOnly) {
         return TransferPanelMainErrorMessage.WITHDRAW_ONLY
       }
 
-      const amountEntered = Number(_amountEntered)
-      const balance = Number(_balance)
+      // No error while loading ERC-20 balance
+      if (selectedTokenBalanceFloat === null) {
+        return undefined
+      }
 
-      if (amountEntered > balance) {
+      // Check amount against ERC-20 balance
+      if (Number(amount) > selectedTokenBalanceFloat) {
         return TransferPanelMainErrorMessage.INSUFFICIENT_FUNDS
       }
+    }
+    // ETH
+    // Check amount against ETH balance
+    else if (Number(amount) > ethBalanceFloat) {
+      return TransferPanelMainErrorMessage.INSUFFICIENT_FUNDS
+    }
 
-      // The amount entered is enough funds, but now let's include gas costs
-      switch (gasSummary.status) {
-        // No error while loading gas costs
-        case 'idle':
-        case 'loading':
-          return undefined
-
-        case 'error':
-          return TransferPanelMainErrorMessage.GAS_ESTIMATION_FAILURE
-
-        case 'success': {
-          if (selectedToken) {
-            // We checked if there's enough tokens above, but let's check if there's enough ETH for gas
-            const ethBalanceFloat = parseFloat(utils.formatEther(ethBalance))
-
-            if (requiredGasFees > ethBalanceFloat) {
-              return TransferPanelMainErrorMessage.INSUFFICIENT_FUNDS
-            }
+    // The amount entered is enough funds, but now let's include gas costs
+    switch (gasSummary.status) {
+      // No error while loading gas costs
+      case 'idle':
+      case 'loading':
+        return undefined
 
-            return undefined
-          }
+      case 'error':
+        return TransferPanelMainErrorMessage.GAS_ESTIMATION_FAILURE
 
-          if (amountEntered + requiredGasFees > balance) {
+      case 'success': {
+        if (selectedToken) {
+          // We checked if there's enough tokens above, but let's check if there's enough ETH for gas
+          if (requiredGasFees > Number(ethBalanceFloat)) {
             return TransferPanelMainErrorMessage.INSUFFICIENT_FUNDS
           }
 
           return undefined
         }
+
+        if (Number(amount) + requiredGasFees > ethBalanceFloat) {
```

Comment:
nit: higher readability and less mental power needed if conditions are extracted into a clearly named var

Reply:
Fixed in https://github.com/OffchainLabs/arbitrum-token-bridge/pull/1257/commits/3b217da728345481e936b5b350b4454d886f6b5c